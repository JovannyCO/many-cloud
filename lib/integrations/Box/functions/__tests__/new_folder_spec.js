describe("new_folder", () => {
  describe("when operation is successfull", () => {
    const expected_resolution = "Some folder info";
    let func, data;
    beforeEach(() => {
      data = {
        client: {
          folders: {
            create: jest.fn().mockImplementation(() => {
              return new Promise((resolve, reject) => {
                resolve(expected_resolution);
              });
            })
          }
        }
      };
      func = require("../new_folder")(data);
    });

    it("resolves with the expected resolution", async () => {
      await expect(func()).resolves.toEqual(expected_resolution);
    });
  });

  describe("when operation is unsuccessfull", () => {
    let func, data;
    beforeEach(() => {
      data = {
        client: {
          folders: {
            create: jest.fn().mockImplementation(() => {
              return new Promise((resolve, reject) => {
                reject(new Error("That didn't work!"));
              });
            })
          }
        }
      };
      func = require("../new_folder")(data);
    });

    it("rejects with the error", async () => {
      await expect(func()).rejects.toMatchSnapshot();
    });
  });
});
