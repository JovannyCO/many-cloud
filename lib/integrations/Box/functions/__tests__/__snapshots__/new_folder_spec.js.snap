// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new_folder when operation is unsuccessfull rejects with the error 1`] = `[Error: That didn't work!]`;
